Set up CI matrix experimental flag and concurrency

NSeydoux opened the pull request.

NSeydoux committed:
Update shared actionsOC

NSeydoux committed:
Set up experimental tag on matrix

NSeydoux committed:
Set up concurrency in CI matrix

NSeydoux committed:
Fix env var names

ThisIsMissEm commented on .github/workflows/audit.yml:
```diff
@@ -7,9 +7,11 @@ on:
 jobs:
   audit:
-    uses: inrupt/javascript-style-configs/.github/workflows/reusable-audit.yml@v1.1.0
+    uses: inrupt/typescript-sdk-tools/.github/workflows/reusable-lint.yml@v1.4.2
```
Copy and paste error: 
```suggestion
    uses: inrupt/typescript-sdk-tools/.github/workflows/reusable-audit.yml@v1.4.2
```

NSeydoux replied:
Oh good catch thanks

ThisIsMissEm commented on .github/workflows/e2e-browser.yml:
```diff
@@ -7,90 +7,49 @@ env:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
 jobs:
-  build:
```
@NSeydoux we actually should move towards this, as otherwise we are building N times when we only need to build once. I know it's inconsistent though across repos. Haven't had a chance to roll out "prebuilding the SDK" across to all repos — this is the whole "one pipeline" dream.

NSeydoux replied:
Oh I see, I missed that. I've added it back with a short comment explaining this.

ThisIsMissEm reviewed (changes requested):
Just needs to fix the audit action due to a copy and paste error.

NSeydoux committed:
Re-enable separate build step

NSeydoux committed:
Fix typo

ThisIsMissEm commented on .github/workflows/e2e-browser.yml:
```diff
@@ -37,60 +36,47 @@ jobs:
-
-      - name: Setup Playwright
-        run: npx playwright install --with-deps
+      - run: npm run build
```
You don't need this in a "build once, test everywhere" setup.

ThisIsMissEm commented on .github/workflows/e2e-browser.yml:
```diff
@@ -37,60 +36,47 @@ jobs:
 
-      - name: Prepare browser-based end-to-end tests
-        run: npm run test:e2e:browser:build
+      - run: npm run test:e2e:browser:build
```
We could also likely do this once for the codebase, and then reuse across tests (FUTURE)

ThisIsMissEm commented on .github/workflows/e2e-node.yml:
```diff
@@ -7,26 +7,31 @@ on:
-        # "Inrupt Dev-Next" is a 2.x environment.
-        environment-name: ["ESS Dev-Next", "NSS"]
+        node-version: [16.x, 14.x]
+        environment-name: ["ESS PodSpaces", "NSS"]
```
```suggestion
#        environment-name: ["ESS PodSpaces", "NSS"]
        environment-name: ["ESS PodSpaces"]
```

Let's de-provision NSS for now, and focus on getting ESS PodSpaces and ESS Dev-Next rolled out as testing environments. We can add this back in across all the SDKs in the future, if we decide to continue testing against NSS directly from here — we need to actually come up with our formal strategy for testing other servers than ESS.

NSeydoux replied:
I agree that we need an overarching strategy, but why remove the NSS tests that are currently implemented and working?

ThisIsMissEm replied:
Because the NSS environment is not managed by our new e2e-environments.

ThisIsMissEm replied:
Though it is incidentally supported here: https://github.com/inrupt/typescript-sdk-tools/blob/main/packages/internal-test-env/index.ts#L34-L39

ThisIsMissEm replied:
Let's get a plan in place for rolling out e2e-environments for NSS and CSS in general, then we can add it back in.

NSeydoux replied:
My concern is that we already have a lot of things to get done before testing on CSS/NSS, so if we disable this now we'll be blind for a while, even when we roll out stuff like the authn refactor, which I'd prefer to be aware of it it breaks NSS compatibility.

NSeydoux committed:
Apply suggestions from code review

Co-authored-by: Emelia Smith <ThisIsMissEm@users.noreply.github.com>

NSeydoux committed:
Add comment about NSS tests

ThisIsMissEm committed:
chore: align e2e-browser tests with other repositories

This does increase the run time of the tests, but consistency across SDKs is better than fast but inconsistent

ThisIsMissEm committed:
Use page steps in series instead of parallel

ThisIsMissEm committed:
Move macos-latest tests to experimental

ThisIsMissEm committed:
Update dependencies in testApp

ThisIsMissEm committed:
chore: upgrade playwright
